homeassistant/components/hive/binary_sensor.py

```diff
@@ -36,17 +37,36 @@ def unique_id(self):
     @property
     def device_info(self):
         """Return device information."""
-        return {"identifiers": {(DOMAIN, self.unique_id)}, "name": self.name}
+        return {
+            "identifiers": {(DOMAIN, self.device["device_id"])},
+            "name": self.device["device_name"],
+            "model": self.device["deviceData"]["model"],
+            "manufacturer": self.device["deviceData"]["manufacturer"],
+            "sw_version": self.device["deviceData"]["version"],
+            "via_device": (DOMAIN, self.device["parentDevice"]),
+        }
 
     @property
     def device_class(self):
         """Return the class of this sensor."""
-        return DEVICETYPE_DEVICE_CLASS.get(self.node_device_type)
+        return DEVICETYPE.get(self.device["hiveType"])["type"]
+
+    @property
+    def icon(self):
+        """Return the class of this sensor."""
+        return DEVICETYPE.get(self.device["hiveType"]).get("icon", None)
```

Comment:
This will error if the first lookup returns None.

None is the default value returned if the key is missing in the dict.